pkg/controller/template/render.go

```diff
@@ -268,15 +271,19 @@ func generateMachineConfigForName(config *RenderConfig, role, name, templateDir,
 		return nil, fmt.Errorf("error transpiling ct config to Ignition config: %v", err)
 	}
 
-	mcfg := MachineConfigFromIgnConfig(role, name, ignCfg)
+	rawIgnCfg, err := json.Marshal(ignCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling Ignition config: %v", err)
+	}
+	mcfg := MachineConfigFromRawIgnConfig(role, name, rawIgnCfg)
 	// And inject the osimageurl here
 	mcfg.Spec.OSImageURL = config.OSImageURL
 
 	return mcfg, nil
 }
 
-// MachineConfigFromIgnConfig creates a MachineConfig with the provided Ignition config
-func MachineConfigFromIgnConfig(role, name string, ignCfg *igntypes.Config) *mcfgv1.MachineConfig {
+// MachineConfigFromRawIgnConfig creates a MachineConfig with the provided Ignition config
+func MachineConfigFromRawIgnConfig(role, name string, rawIgnCfg []byte) *mcfgv1.MachineConfig {
```

Comment:
Why not move the `json.Marshal(ignCfg)` call inside this function? Every single caller does that immediately before, and it would be more type-safe.

Reply:
This function has to work with Ignition Config Spec 2 as well as 3 (meaning in input cannot be a typed spec 2 Ignition config).

It could probably take any interface as arg and then marshal that internally, but that wouldn't make it any more type safe.

Reply:
AFAICS there are 5 callers that all start with a `ctrlcommon.NewIgnConfig()` (one of them updating the value), and one caller right above that comes from the updated `transpileToIgn`. All of them are
```go
*igntypes.Config // igntypes "github.com/coreos/ignition/config/v2_4/types"
```
aren’t they?

(And even if it weren’t any more type-safe, the 6 repeated `json.Marshal` calls could be avoided, and if this function _wanted_ to do some kind of sanity check on the contents, it would probably be cheaper on a Go value than on a `[]byte`.)

Reply:
The RawExt conversion is a prerequisite for supporting both v2 and v3 Ignition configs.
I've changed the function to take `interface{}` as arg and do the marshalling within.
Please refer to #996 (and specifically https://github.com/openshift/machine-config-operator/pull/996/files#diff-7bee216e56f15d868e9558dedc3fda0eR283) for all things related to the RawExt conversion.